### PR TITLE
Port Xilem and Masonry to winit 0.30.0

### PR DESCRIPTION
**Pull request by xorgy**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -70,4 +70,4 @@ bitflags = "2"
 tracing = "0.1.37"
 fnv = "1.0.7"
 instant = { version = "0.1.6", features = ["wasm-bindgen"] }
-winit = "0.29"
+winit = { version = "0.30", features = ["rwh_05"]}
```
Comment (waywardmonkeys) [inline Cargo.toml:73]: This `rwh_05` isn't needed.

**Review by waywardmonkeys — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (xorgy) [reply Cargo.toml:73]: Ope, yeah had to add that to figure something out, but not needed now.

**Comment by waywardmonkeys** [comment]
How hard would it be to also do the same for the new Masonry-based stuff?

I'd started looking at it, I think, but got busy with other stuff for work ...

**Comment by xorgy** [comment]
> How hard would it be to also do the same for the new Masonry-based stuff?

I have no idea, that's something to ask Daniel. Right now I'm doing this to simplify the AccessKit integration in classic Xilem.

**Comment by xorgy** [comment]
We could pin masonry winit to `0.29.0` for now if that's helpful.

**Comment by xorgy** [comment]
Looks like there are four examples directly interacting with winit event loops, and also `crates/masonry/src/event_loop_runner.rs`

**Comment by waywardmonkeys** [comment]
I just suspect that a bare-bones port might be simple enough to just do it at all at the same time and not bloat CI times by having to compile multiple versions of all sorts of things ... If it is, then great. If it isn't, then we could just keep it as it is where one thing is using 0.30 and the other is still on 0.29.

**Comment by xorgy** [comment]
I have a quick and dirty port of masonry event_loop_runner, just testing it out now

File: Cargo.toml [hunk 2]
```diff
@@ -70,4 +70,4 @@ bitflags = "2"
 tracing = "0.1.37"
 fnv = "1.0.7"
 instant = { version = "0.1.6", features = ["wasm-bindgen"] }
-winit = "0.29"
+winit = "0.30"
```
Comment (waywardmonkeys) [inline Cargo.toml:73]: Might as well switch to a workspace dependency now.

File: crates/masonry/src/event_loop_runner.rs [hunk 3]
```diff
@@ -5,123 +5,169 @@ use tracing::warn;
...
+                }
+                render_root::RenderRootSignal::SetCursor(cursor) => {
+                    self.window.set_cursor(cursor);
+                }
+                render_root::RenderRootSignal::SetSize(size) => {
+                    // TODO - Handle return value?
+                    let _ = self.window.request_inner_size(size);
+                }
+                render_root::RenderRootSignal::SetTitle(title) => {
+                    self.window.set_title(&title);
```
Comment (waywardmonkeys) [inline crates/masonry/src/event_loop_runner.rs:167]: These should probably use user events in winit, but that's for the future, not today.

**Review by waywardmonkeys — CHANGES_REQUESTED** [review]
As long as others are okay with the Masonry specific parts, I'm okay with the rest for a first pass. Just should change the one dependency to be a workspace dep.

**Comment by PoignardAzur** [comment]
LGTM.

File: crates/masonry/src/event_loop_runner.rs (hunk 3 above)
Comment (xorgy) [reply crates/masonry/src/event_loop_runner.rs:166]: I think Masonry signals serve a different purpose.

**Review by DJMcNab — APPROVED** [review]
I have *not* scanned through this, but am just formalizing the two other approvals

**Review by waywardmonkeys — APPROVED** [review]

**Pull request merged**